oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -176,6 +181,64 @@ public void getDefaultCredentials_noCredentials_singleGceTestRequest() {
         ComputeEngineCredentials.MAX_COMPUTE_PING_TRIES);
   }
 
+  @Test
+  public void getDefaultCredentials_noCredentials_linuxNotGce() throws IOException {
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    testProvider.setProperty("os.name", "Linux");
+    String productFilePath = SMBIOS_PATH_LINUX;
+    InputStream productStream = new ByteArrayInputStream("test".getBytes());
+    testProvider.addFile(productFilePath, productStream);
+
+    assertFalse(ComputeEngineCredentials.checkStaticGceDetection(testProvider));
+  }
+
+  @Test
+  public void getDefaultCredentials_static_linux() throws IOException {
```

Comment:
These tests could be simplified to a single test with a loop going over a map of inputs.

Reply:
in this case explicit test cases preferred for readability

Reply:
I think it will still be easy to understand if we have keyvalue pair of input and expected output. But up to you.